mmv1/third_party/terraform/tests/resource_compute_target_ssl_proxy_test.go.erb

```diff
@@ -25,22 +35,34 @@ func TestAccComputeTargetSslProxy_update(t *testing.T) {
 			{
 				Config: testAccComputeTargetSslProxy_basic1(target, sslPolicy, cert1, backend1, hc),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckComputeTargetSslProxy(
-						t, "google_compute_target_ssl_proxy.foobar", "NONE", cert1),
+					testAccCheckComputeTargetSslProxyExists(
+						t, "google_compute_target_ssl_proxy.foobar", &proxy),
+					testAccCheckComputeTargetSslProxyHeader(t, "NONE", &proxy),
+					testAccCheckComputeTargetSslProxyHasSslCertificate(t, cert1, &proxy),
 				),
 			},
 			{
 				Config: testAccComputeTargetSslProxy_basic2(target, sslPolicy, cert1, cert2, backend1, backend2, hc),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckComputeTargetSslProxy(
-						t, "google_compute_target_ssl_proxy.foobar", "PROXY_V1", cert2),
+					testAccCheckComputeTargetSslProxyExists(
+						t, "google_compute_target_ssl_proxy.foobar", &proxy),
+					testAccCheckComputeTargetSslProxyHeader(t, "PROXY_V1", &proxy),
+					testAccCheckComputeTargetSslProxyHasSslCertificate(t, cert2, &proxy),
+				),
+			},
+			{
+				Config: testAccComputeTargetSslProxy_certificateMap(resourceSuffix),
```

Comment:
Can we test the update function of `certificate_map` similar to how the `ssl_certificates` field is tested?

Reply:
Could you elaborate? What exactly does "similar" mean? 

The only difference that I see is that in `certificate_map` case we use different resource name format than in `ssl_certificates`:

```
certificate_map = "//certificatemanager.googleapis.com/${google_certificate_manager_certificate_map.map.id}"

vs

ssl_certificates = [google_compute_ssl_certificate.bar.name]
```

I'm not sure whether this format is necessary, I've seen it in one of tests though: `mmv1/third_party/terraform/tests/resource_compute_target_https_proxy_test.go.erb`

Reply:
BTW: I no longer have these errors about missing credentials. I must have fixed while working on the task. `TestAccComputeTargetSslProxy_update` test passes.

Reply:
Sorry for the ambiguity. In `testAccComputeTargetSslProxy_basic1`, `ssl_certificates` is set to `google_compute_ssl_policy.foo.self_link`. It is updated to `google_compute_ssl_certificate.bar.name` in `testAccComputeTargetSslProxy_basic2`. 

I wanted to test out the update functionality of `certificate_map` (especially as it uses a `update_verb` and separate `update_url`) by adding a step where we update the value. However you think that should be done

Reply:
Done (similar to `testAccComputeTargetSslProxy_basic1` and `testAccComputeTargetSslProxy_basic2`)